### PR TITLE
Annotations extra require - sagemaker

### PR DESCRIPTION
**Pull request by wild-endeavor**

### CODE REVIEW

**Review by katrogan — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -8,13 +8,15 @@
     "sidecar": ["k8s-proto>=0.0.3,<1.0.0"],
     "schema": ["numpy>=1.14.0,<2.0.0", "pandas>=0.22.0,<2.0.0", "pyarrow>=0.11.0,<1.0.0"],
     "hive_sensor": ["hmsclient>=0.0.1,<1.0.0"],
+    "sagemaker": ["sagemaker-training>=3.6.2,<4.0.0"],
     "all-spark2.4": [
```
Comment (EngHabu) [inline setup.py:12]: This is very annoying to have... imagine then having `all-spark2.4-pyarrow1.0` ...

**Review by EngHabu — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (kumare3) [reply setup.py:12]: I think this is done wrongly

Comment (wild-endeavor) [reply setup.py:12]: @akhurana001 when did we say we were gonna deprecate 2.x support entirely?

Comment (akhurana001) [reply setup.py:12]: Have https://github.com/lyft/flyte/issues/483 for the same. I can make a PR if we are ok doing that ?

**Pull request merged**